apps/project/dataloaders.py

```diff
@@ -216,6 +217,55 @@ def resolve(self):
         )
 
 
+class ProjectSummaryStatLoader(WithContextMixin):
+    def get_stat(self):
+        projects = Project.get_for_member(self.context.request.user)
+        # Lead stats
+        leads = Lead.objects.filter(project__in=projects)
+        total_leads_tagged_count = leads.annotate(entries_count=models.Count('entry')).filter(entries_count__gt=0).count()
+        total_leads_tagged_and_controlled_count = leads.annotate(
+            controlled_entries_count=models.Count(
+                'entry', filter=models.Q(entry__controlled=True)
+            )
+        ).filter(controlled_entries_count__gt=0).count()
+        # Entries activity
+        recent_projects_id = list(
+            projects.annotate(
+                entries_count=Cast(KeyTextTransform('entries_activity', 'stats_cache'), models.IntegerField())
+            ).filter(entries_count__gt=0).order_by('-entries_count').values_list('id', flat=True)[:3])
+        recent_entries = Entry.objects.filter(
+            project__in=recent_projects_id,
+            created_at__gte=(timezone.now() + relativedelta(months=-3))
+        )
+        recent_entries_activity = {
+            'projects': (
+                recent_entries.order_by().values('project')
+                .annotate(count=models.Count('*'))
+                .filter(count__gt=0)
+                .values('count', 'project')
+            ),
+            'activities': (
+                recent_entries.order_by('project', 'created_at__date').values('project', 'created_at__date')
+                .annotate(count=models.Count('*'))
+                .values('project', 'count', date=models.Func(models.F('created_at__date'), function='DATE'))
```

Comment:
@AdityaKhatri @subinasr 
'projects' data seems to be redundant, as 'activities' includes them here. 
Are we using them both in the frontend? Can frontend calculate the `projects` data from `activities`?
If yes, we could remove redundant calculation from backend.

Reply:
Yes, this can be calculated in the frontend. We can skip this and only have activities.